q2_types/feature_data/_format.py

```diff
@@ -90,38 +90,59 @@ class TSVTaxonomyFormat(model.TextFileFormat):
 
     Optionally followed by other arbitrary columns.
 
-    This format supports comment lines starting with #, and blank lines. The
-    expected header must be the first non-comment, non-blank line. In addition
-    to the header, there must be at least one line of data.
+    This format supports blank lines. The expected header must be the first
+    non-blank line. In addition to the header, there must be at least one line
+    of data.
 
     """
     HEADER = ['Feature ID', 'Taxon']
 
-    def sniff(self):
+    def _check_n_records(self, n=None):
         with self.open() as fh:
             data_lines = 0
             header = None
-            while data_lines < 10:
-                line = fh.readline()
 
+            file_ = enumerate(fh) if n is None else zip(range(n), fh)
+
+            for i, line in file_:
+                # Tracks line number for error reporting
+                i = i + 1
                 if line == '':
                     # EOF
                     break
                 elif line.lstrip(' ') == '\n':
                     # Blank line
                     continue
 
-                cells = line.rstrip('\n').split('\t')
+                cells = line.strip('\n').split('\t')
+
                 if header is None:
                     if cells[:2] != self.HEADER:
-                        return False
+                        raise ValidationError("['Feature ID' and 'Taxon'] "
+                                              "must be the first two header "
+                                              "values to be valid.\n\n The "
+                                              "first two header values "
+                                              "provided are: {}.\nIssue on "
+                                              "line {}"
+                                              .format(cells[:2], i))
                     header = cells
                 else:
                     if len(cells) != len(header):
-                        return False
+                        raise ValidationError("Number of columns are not the "
```

Comment:
Please move the offending line number up front in this error message (similar to my proposed change, above).